### PR TITLE
response to shutdown signals

### PR DESCRIPTION
**Pull request by phosae**
image

the server now can be terminated by Ctrl-C. solve this problem in this PR.

### CODE REVIEW

**Comment by phosae** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cmd/picoshare/main.go [hunk 1]
```diff
@@ -52,15 +55,22 @@ func main() {
...
 
 	port := os.Getenv("PORT")
 	if port == "" {
 		port = "4001"
 	}
-	log.Printf("listening on %s", port)
 
-	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+	stop := setupSignalHandler()
+	httpSrv := &http.Server{Addr: fmt.Sprintf(":%s", port), Handler: h}
```
Comment (mtlynch) [inline cmd/picoshare/main.go:65]: Why `&http.Server` instead of just `http.Server`?

File: cmd/picoshare/main.go [hunk 2]
```diff
@@ -82,3 +92,16 @@ func ensureDirExists(dir string) {
 func isLitestreamEnabled() bool {
 	return os.Getenv("LITESTREAM_BUCKET") != ""
 }
+
+func setupSignalHandler() (stopCh <-chan struct{}) {
```
Comment (mtlynch) [inline cmd/picoshare/main.go:96]: Can we delete the name `stopCh`? The rest of the code doesn't use named returns, and we don't use `stopCh` in this function.

**Review by mtlynch — CHANGES_REQUESTED** [review]
Awesome, thanks! Just a few minor questions.

**Review by mtlynch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**